### PR TITLE
add a note about gem metadata and native gems

### PR DESCRIPTION
**Pull request by lamont-granquist**
native gems can't be installed here.

### CODE REVIEW

File: chef_master/source/config_rb_metadata.rst [hunk 1]
```diff
@@ -115,6 +115,10 @@ This configuration file has the following settings:
...
+
+   This feature is not intended to be a general purpose replacement for the ``chef_gem`` resource, and does not internally re-use the ``chef_gem`` resource.  It targets a specific use case of external chef libraries shipped as gems (such as the "chef-sugar" and "poise" gems).  These gems require eager loading so that their features are accessable very early in the chef-client run so that they can be used in library files and attributes files.  Delaying their installation until recipe time would mean that on first run none of their features could be used in libraries or attribute files.  Those gems could not use the ``chef_gem`` resource, so this mechanism was created to allow them to install very early, with the understanding that it had an inherent limitation to being unable to install native gems.  To the extent that other pure ruby gems can be installed this way they are free to, but this mechanism was not designed to suit all use cases and cannot be adapted to all use cases.
```
Comment (mjingle) [inline chef_master/source/config_rb_metadata.rst]: @lamont-granquist Can you please clarify what "this feature" is exactly? Thanks!

Comment (lamont-granquist) [reply chef_master/source/config_rb_metadata.rst]: the gem metadata feature.

Comment (lamont-granquist) [reply chef_master/source/config_rb_metadata.rst]: ah i see, i reused that word meaning different things in different contexts.  fixing...

File: chef_master/source/config_rb_metadata.rst [hunk 2]
```diff
@@ -115,6 +115,10 @@ This configuration file has the following settings:
...
+
+   The gem metadata.rb setting is not intended to be a general purpose replacement for the ``chef_gem`` resource, and does not internally re-use the ``chef_gem`` resource.  It addresses a specific use case of external chef libraries shipped as gems (such as the "chef-sugar" and "poise" gems).  These gems require eager loading so that their features are accessable very early in the chef-client run so that they can be used in library files and attributes files.  Delaying their installation until recipe time would mean that on first run none of their features could be used in libraries or attribute files.  Those gems could not use the ``chef_gem`` resource, so this mechanism was created to allow them to install very early, with the understanding that it had an inherent limitation to being unable to install native gems.  To the extent that other pure ruby gems can be installed this way they are free to, but this mechanism was not designed to suit all use cases and cannot be adapted to all use cases.
```
Comment (mjingle) [inline chef_master/source/config_rb_metadata.rst]: @lamont-granquist Is the "mechanism" mentioned in line 120 the same as the `gem metadata.rb setting` or is this referring to something else? Thank you for the clarification in advance!

Comment (lamont-granquist) [reply chef_master/source/config_rb_metadata.rst]: yes "this mechanism" means "gem metadata.rb setting"

Comment (lamont-granquist) [reply chef_master/source/config_rb_metadata.rst]: (both cases)

Comment (mjingle) [reply chef_master/source/config_rb_metadata.rst]: Awesome, thanks!

**Review by mjingle — APPROVED** [review]
tenor-242800872

**Comment by kagarmoe** [comment]
tenor-33211239

**Pull request merged**